pkgs/development/python-modules/scikit-image/default.nix

```diff
@@ -41,14 +48,51 @@ buildPythonPackage rec {
     tifffile
   ];
 
-  checkInputs = [ pytest ];
+  checkInputs = [ pytestCheckHook ];
+
+  # (1) The package has cythonized modules, whose .so libs will appear only in the wheel, i.e. in nix store;
+  # (2) To stop Python from importing the wrong directory, i.e. the one in the build dir, not the one in nix store, `skimage` dir should be removed or renamed;
+  # (3) Therefore, tests should be run on the installed package in nix store.
 
-  # No tests in archive
-  doCheck = false;
+  # See e.g. https://discourse.nixos.org/t/cant-import-cythonized-modules-at-checkphase/14207 on why the following is needed.
+  preCheck = ''
+    rm -r skimage
+  '';
```

Comment:
I just packaged another cython package, and didn't need this in the end. [Commit here](https://github.com/samuela/nixpkgs/commit/8fda9a49a7b0bf6159d2bb46c64022cffe59dce0). Have you tried removing this now that pytest is being pointed towards `installedPackageRoot`?

Reply:
Well, in this case it still breaks pytest, specifically test collection:
```
E   _pytest.pathlib.ImportPathMismatchError: ('skimage.conftest', '/nix/store/sj4bdpny8w0kh7dqvgbnz7asgi7rjg93-python3.8-scikit-image-0.18.3/lib/python3.8/site-packages/skimage/conftest.py', PosixPath('/build/source/skimage/conftest.py'))
```

I think it depends on an actual test configuration. Here it's probably easier to leave it as is.